src/util/ProjectIO.as

```diff
@@ -207,48 +207,113 @@ public class ProjectIO {
 		}
 	}
 
+	// Load all images in all costumes from their image data, then call whenDone.
 	public function decodeAllImages(objList:Array, whenDone:Function, fail:Function = null):void {
-		// Load all images in all costumes from their image data, then call whenDone.
-		function imageDecoded():void {
-			for each (var o:* in imageDict) {
-				if (o == 'loading...') return; // not yet finished loading
+		// This should be called on success or failure of each image
+		function imageDone():void {
+			if (--numImagesToDecode == 0) {
```

Comment:
Sorry, I'm not sure I totally grok the design patterns in our AS3 code. Totally ok to keep if this is normal. 
Nit: this feels a tad bit hard to scan for me without a performance benefit AFAIK.